### DISCUSSION TITLE
How to access "op" and "in" fields in lottie animation?

### DISCUSSION THREAD

**Discussion by rismirnov** [LottieFiles/dotlottie-web#212] [Q&A]
I would like to be able to get the outPoint(the frame at which the Lottie animation ends at) and inPoint(the frame at which the Lottie animation starts at) value for a lottie animation. Based on these values ​​I want to set the segment to be played by
```player.setSegment(startFrame, endFrame)```. Tell me, is there any way to do this?

![изображение](https://github.com/LottieFiles/dotlottie-web/assets/53395925/386247b5-cc34-4987-919d-68cb5eecfed6)

**Comment by theashraf** [comment]
You may fetch the Lottie file first, extract the `op` and `in` values, and then pass them to the DotLottie player for rendering with the segments

```js
import { DotLottie } from "@lottiefiles/dotlottie-web";

const animationUrl = "https://lottie.host/path/to/animation.json";

fetch(animationUrl)
  .then((res) => res.json())
  .then((animationData) => {
    const op = animationData.layers[0].op;
    const in = animationData.layers[0].in;

    const dotLottie = new DotLottie({
      container: document.querySelector("#dotLottie-canvas"),
      data: animationData,
      loop: true,
      autoplay: true,
      segment: [in, op],
    });
  })
  .catch((error) => console.error("Error loading animation:", error));
  ```
